xds/server.go

```diff
@@ -332,6 +335,8 @@ func (s *GRPCServer) handleServingModeChanges(updateCh *buffer.Unbounded) {
 // corresponding pending RPCs on the client side will get notified by connection
 // errors.
 func (s *GRPCServer) Stop() {
+	s.clientMu.Lock()
+	defer s.clientMu.Unlock()
```

Comment:
Would it work if we hold the lock only around `s.quit.Fire()`? Same comment applies to `GracefulStop` as well.

Reply:
I don't think so. If you don't protect xdsC and xdsClientClose in the mutex, you can get a write in initXDSClient() and a read in Stop(), which gives you corrupted memory. Also, I think it makes sense to do this processing of the whole (client create/increase ref) and (decrease ref client close if set)  atomically, serially rather than interspliced.

Reply:
My only concern was that we are holding the lock when calling `gs.Stop()` which is a blocking call. But I guess its OK, since the xDS enabled server is being closed at this point, and its not a big deal if we hold the lock and block while the underlying grpc server is being closed.

Reply:
Ok, sounds good. Yeah blocking a simultaneous Serve() call seems ok, especially since in other components we process whole events serially through patterns such as a run() goroutine.